graphstorm-processing/graphstorm_processing/distributed_executor.py

```diff
@@ -0,0 +1,342 @@
+        gsprocessing_args.input_prefix = str(
+            Path(gsprocessing_args.input_prefix).resolve(strict=True)
+        )
+        if not Path(gsprocessing_args.output_prefix).absolute().exists():
+            os.makedirs(gsprocessing_args.output_prefix)
+        gsprocessing_args.output_prefix = str(
+            Path(gsprocessing_args.output_prefix).resolve(strict=True)
+        )
+        filesystem_type = "local"
+
+    # local input location for config file and execution script
+    if is_sagemaker_execution:
+        local_config_path = "/opt/ml/processing/input/data"
+    else:
+        # If not on SageMaker, assume that we are running in a
+        # native env with local input or Docker execution with S3 input
+        if filesystem_type == "local":
+            local_config_path = gsprocessing_args.input_prefix
+        else:
+            tempdir = tempfile.TemporaryDirectory()  # pylint: disable=consider-using-with
+            input_bucket = gsprocessing_args.input_prefix.split("/")[2]
+            input_s3_prefix = gsprocessing_args.input_prefix.split("/", 3)[3]
+            s3 = boto3.client("s3")
+            s3.download_file(
+                input_bucket,
+                f"{input_s3_prefix}/{gsprocessing_args.config_filename}",
+                os.path.join(tempdir.name, gsprocessing_args.config_filename),
+            )
+            local_config_path = tempdir.name
+
+    # local output location for metadata files
+    if is_sagemaker_execution:
+        local_output_path = "/opt/ml/processing/output"
+    else:
+        if filesystem_type == "local":
+            local_output_path = gsprocessing_args.output_prefix
+        else:
+            # Only needed for local execution with S3 data
+            local_output_path = tempdir.name
+
+    if gsprocessing_args.num_output_files == "" or gsprocessing_args.num_output_files is None:
+        gsprocessing_args.num_output_files = -1
+
+    executor_configuration = ExecutorConfig(
+        local_config_path=local_config_path,
+        local_output_path=local_output_path,
+        input_prefix=gsprocessing_args.input_prefix,
+        output_prefix=gsprocessing_args.output_prefix,
+        num_output_files=gsprocessing_args.num_output_files,
+        config_filename=gsprocessing_args.config_filename,
+        sm_execution=is_sagemaker_execution,
+        filesystem_type=filesystem_type,
+        add_reverse_edges=gsprocessing_args.add_reverse_edges,
+    )
+
+    dist_executor = DistributedExecutor(executor_configuration)
+
+    dist_executor.run()
+
+    # Save arguments to file for posterity
+    with open(os.path.join(local_output_path, "launch_arguments.json"), "w", encoding="utf-8") as f:
+        json.dump(dataclasses.asdict(gsprocessing_args), f, indent=4)
+
+    # Used during local execution with S3 data, uploading arguments file to S3
+    if not is_sagemaker_execution and filesystem_type == "s3":
```

Comment:
What if it is sagemaker execution?

Reply:
In SageMaker execution, the arguments file is saved under `/opt/ml/processing/output` so SageMaker will take care of uploading it once the job is done.

Reply:
Can you add some comments here to explain the difference between SageMaker execution and local execution?